### PR TITLE
rename reorgEventChannel to ChainHeadChannel

### PR DESCRIPTION
**Pull request by rolfyone**
- ChainHeadChannel will also notify of a new chain head, but initially the intent is to replace the Reorg functionality currently in use.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

## Documentation

- [X] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/util/BeaconStateUtil.java [hunk 1]
```diff
@@ -464,6 +464,10 @@ public static UInt64 compute_start_slot_at_epoch(UInt64 epoch) {
     return epoch.times(SLOTS_PER_EPOCH);
   }
 
+  public static boolean is_start_slot_of_epoch(final UInt64 slot) {
+    return slot.mod(SLOTS_PER_EPOCH).equals(UInt64.ZERO);
+  }
```
Comment (ajsutton) [inline ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/util/BeaconStateUtil.java]: This doesn't appear to be used?

File: storage/src/main/java/tech/pegasys/teku/storage/client/RecentChainData.java [hunk 2]
```diff
@@ -260,14 +262,18 @@ private void updateChainHead(
               commonAncestorSlot);
         }
 
+        final UInt64 epoch = compute_epoch_at_slot(newChainHead.getSlot());
+        final boolean epochTransition =
+            newChainHead.getSlot().equals(compute_start_slot_at_epoch(epoch));
```
Comment (ajsutton) [inline storage/src/main/java/tech/pegasys/teku/storage/client/RecentChainData.java]: I don't think this is right.  We might wind up skipping multiple slots and so miss the first slot of the epoch.  I think it should check if the forkChoiceSlot from previousChainHead is from an earlier epoch than the forkChoiceSlot from newChainHead.

**Review by ajsutton — COMMENTED** [review]
Looks good generally, just need to handle skipping slots when setting the epoch transition flag.

File: ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/util/BeaconStateUtil.java (hunk 1 above)
Comment (rolfyone) [reply ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/util/BeaconStateUtil.java]: at the moment its only in test code but the `RecentChainData.updateChainHead` will call it so i dont have to code the decision into that...

File: storage/src/main/java/tech/pegasys/teku/storage/client/RecentChainData.java (hunk 2 above)
Comment (rolfyone) [reply storage/src/main/java/tech/pegasys/teku/storage/client/RecentChainData.java:267]: hrm that's the other place i should be using the other function - then i dont need most of it, just pass into the function `is_start_slot_of_epoch`

Comment (ajsutton) [reply storage/src/main/java/tech/pegasys/teku/storage/client/RecentChainData.java]: The point is that you're not looking for the first slot in the epoch.  You might be going from slot 30 to 50 in one update and that's an epoch transition even though you never got to slot 32.

**Review by ajsutton — APPROVED** [review]
LGTM.

**Pull request merged**